### ISSUE TITLE
Readxl +Dates are missing 1 second

### ISSUE THREAD

**Issue by SalmaSallem** [tidyverse/readxl#315]
Hi,

I'm trying to read an excel file using Readxl, It's very fast and I like it. Unless, I got a problem with the dates values: Dates are all missing 1 second in the data frame. 
Example: 2017-01-01 00:15:00 became 2017-01-01 00:14:59
(Joined an example of spreadsheet, and run df = readxl::read_excel(fileName, sheet = 1))
Someone could help with this please?
Thank you
[test.xlsx](https://github.com/tidyverse/readxl/files/890898/test.xlsx)

**Comment by jennybc** [comment]
Can you tell me where this xlsx came from? Did Excel write it?

**Comment by jennybc** [comment]
I think is a printing issue.

I truncated your sheet to the top 2 rows for brevity. Two rows are enough to demonstrate your point.

The first column is exactly as you sent. In the second column, I apply a more detailed date format to your data (`yyyy-mm-dd h:mm:ss.000`). The third column shows the underlying floating point number Excel uses to represent date times.

<img width="552" alt="screen shot 2017-04-04 at 10 13 41 pm" src="https://cloud.githubusercontent.com/assets/599454/24690776/018594a4-1984-11e7-9405-6a1c6c29f50c.png">

The above reveals that the second time is not actually at 00:15, but is rather several milliseconds prior to 00:15. With your date format this is hidden. In terms of printing, Excel is rounding to the nearest second (which seems sensible).

Now I read this sheet into R with `read_excel()`, allow it print with default settings, and then apply a similarly detailed date format:

``` r
library(readxl)
(xlsx <- read_excel("iss315_xlsx.xlsx"))
#> # A tibble: 2 × 2
#>                  Date `Date as number`
#>                <dttm>            <dbl>
#> 1 2017-01-01 00:15:00         42736.01
#> 2 2017-01-02 00:14:59         42737.01
op <- options(digits.secs = 3)
xlsx
#> # A tibble: 2 × 2
#>                      Date `Date as number`
#>                    <dttm>            <dbl>
#> 1 2017-01-01 00:15:00.000         42736.01
#> 2 2017-01-02 00:14:59.994         42737.01
options(op)
```

With default printing, it looks like R has lost a second off the 2nd time. But the more detailed printing shows we are working with the same data, at least to 1/100th or 1/1000th of a second.

Here's a relevant StackOverflow thread on R's POSIXct printing:

<http://stackoverflow.com/questions/7726034/how-r-formats-posixct-with-fractional-seconds>

Conversion of Excel datetimes to R's POSIXct dates is legitimately fussy and I don't claim readxl is perfect. But I can't find a modification that "works" for your sheet that doesn't break the date import for other test sheets in more meaningful ways.

If you can show readxl brings in the wrong data in terms of downstream **computations**, then that is a bug and we'd like to know. But for now I consider this a date time printing inconsistency between Excel and R.

cc @hadley

**Comment by SalmaSallem** [comment]
Hi Jenny,

Thanks for your answer. it's really helpful. 
As an answer to your first question we got the excel from another party, so no idea how they are generated. I just copy paste the dates in the test example. But, the idea is to run the script on multiple spreadsheet coming from different sources, and it will be difficult to check times on each file.
Based on your answer, I added a new operation to the script, and round minutes (in my case, I don't need seconds):
dfResult = mutate (dfResult, date_local = format(round(date_local, units="mins"), format="%Y-%m-%d %H:%M"))
The new result is: 2017-02-01 01:00

Thanks again for your help :)
